code42_connector.py

```diff
@@ -225,11 +229,91 @@ def _handle_remove_high_risk_tag(self, param, action_result):
         )
         return action_result.set_status(phantom.APP_SUCCESS, message)
 
+    """ALERTS ACTIONS"""
+
+    @action_handler_for("get_alert_details")
+    def _handle_get_alert_details(self, param, action_result):
+        alert_id = param["alert_id"]
+        response = self._client.alerts.get_details([alert_id])
+        alert = response.data["alerts"][0]
+        action_result.add_data(alert)
+        action_result.update_summary(
+            {"username": alert["actor"], "user_id": alert["actorId"]}
+        )
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    @action_handler_for("search_alerts")
+    def _handle_search_alerts(self, param, action_result):
+        username = param.get("username")
+        start_date = param.get("start_date")
+        end_date = param.get("end_date")
+        alert_state = param.get("alert_state")
+
+        if not username and not start_date and not end_date and not alert_state:
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                "Code42: Must supply a search term when calling action 'search_alerts`.",
+            )
+
+        if not self._validate_date_range(start_date, end_date):
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                "Code42: Start Date and End Date are both required to search by date range for action 'search_alerts'.",
+            )
+        try:
+            query = self._build_alerts_query(
+                username, start_date, end_date, alert_state
+            )
+        except ValueError as exception:
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                f"Code42: Start Date and End Date must be in format YYYY-mm-dd: {exception} for action 'search_alerts'",
+            )
+
+        response = self._client.alerts.search(query)
+        action_result.add_data(response.data)
+        action_result.update_summary({"total_count": response["totalCount"]})
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    @action_handler_for("set_alert_state")
+    def _handle_set_alert_state(self, param, action_result):
+        alert_id = param["alert_id"]
+        alert_state = param["alert_state"]
+        note = param.get("note")
+        response = self._client.alerts.update_state(alert_state, [alert_id], note=note)
+        action_result.add_data(response.data)
```

Comment:
Getting this output from compile_app.pyc for the `set_alert` JSON definition, it seems to be complaining that I'm not exposing any properties on `action_result.data` in the output array:
```    
set alert state
      Following required data paths not in output list
        action_result.data
```
But the `py42.alerts.update_state` method doesn't actually return any response data, and there's nothing that I've calculated in the action itself that isn't already present in the parameters. It feels bad to programmatically add something useless to `action_result.add_data`. But maybe it's worth it to suppress the compiler warning?

Thoughts?

Reply:
What does the generated action JSON look like? Does it have any sort of placeholder for `.data`?

Reply:
Spun up a dummy action and found that the auto-generated action JSON also has this issue
```    search
      Action name "search" is invalid. Action names should be 2 or 3 words long separated by a space
      Data Path "action_result.parameter.username" has invalid contains, A contains list was provided, but it was empty.
      Data Path "action_result.parameter.state" has invalid contains, A contains list was provided, but it was empty.
      Following required data paths not in output list
        action_result.data
        action_result.summary
```